code/game/objects/items/devices/radio/intercom.dm

```diff
@@ -40,9 +40,11 @@
 
 /obj/item/device/radio/intercom/attack_ai(mob/user as mob)
 	add_hiddenprint(user)
-	add_fingerprint(user)
-	spawn (0)
-		attack_self(user)
+	if(isrobot(user))
+		if(user.client && user.client.eye == user)
+			return attack_hand(user)
```

Comment:
calling attack_self directly was probably better because the only difference is 1 less call and the robots won't leave remote fingerprints

Reply:
fixed